### ISSUE TITLE
["finally"] is better written in dot notation  dot-notation

### ISSUE THREAD

**Issue by dciccale** [eslint/eslint#4580]
`finally` is an es6 reserved keyword, shouldn't I be using it like `["finally"]` ?

UPDATE

1 ESLint version 1.5.1

2 code
```js
modalInstance.result['finally'](function () {
            modalInstance = null;
            if ($state.$current.name === stateName) {
              $state.go('^');
            }
          }
```

3 relevant config

```
    "dot-notation": 1,
```

**Comment by eslintbot** [comment]
@dciccale Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
What's your configuration for the `dot-notation` rule? Is it allowing keywords or not?

Modern browsers can handle reserved words as property names, even in dot notation. Unless you have `allowKeywords: false` configured, I think the rule is working as designed and that you might be making an incorrect assumption about the need to use bracket notation for that property.

**Comment by dciccale** [comment]
ok, I'll use it with dot notation, didn't know that we should do that even if it is a reserved word. thx

**Comment by nzakas** [comment]
Confirmed, working as intended. You can do `.finally` as of ES5.
